src/collections/fifo_mt.c

```diff
@@ -0,0 +1,150 @@
+//
+// Copyright (c) 2024 ZettaScale Technology
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+//
+// Contributors:
+//   ZettaScale Zenoh Team, <zenoh@zettascale.tech>
+//
+
+#include "zenoh-pico/collections/fifo_mt.h"
+
+#include "zenoh-pico/protocol/core.h"
+#include "zenoh-pico/utils/logging.h"
+
+/*-------- Fifo Buffer Multithreaded --------*/
+int8_t _z_fifo_mt_init(_z_fifo_mt_t *fifo, size_t capacity) {
+    int8_t res = _z_fifo_init(&fifo->_fifo, capacity);
+    if (res) {
+        return res;
+    }
+
+#if Z_FEATURE_MULTI_THREAD == 1
+    res = zp_mutex_init(&fifo->_mutex);
```

Comment:
all the zp_* system calls are deprecated, you should use z_mutex & z_condvar instead

Reply:
I don't see them in this branch :(